### PR TITLE
feat:configure secrets and app insight

### PR DESCRIPTION
**Pull request by tomaszpowroznik**
test

### CODE REVIEW

File: src/main/server.ts [hunk 1]
```diff
@@ -1,17 +1,27 @@
...
 import { app } from './app'
 
-const logger = Logger.getLogger('server')
+const logger = Logger.getLogger('server');
 
 // TODO: set the right port for your application
-const port: number = parseInt(process.env.PORT, 10) || 3000
+const port: number = parseInt(process.env.PORT, 10) || 3000;
+
+if (config.secrets && config.secrets.adoption.AppInsightsInstrumentationKey) {
```
Comment (mounikahmcts) [inline src/main/server.ts]: I prefer moving secrets stuff to another file just to keep everything in one place and export here something like setSecrets.

File: src/main/server.ts [hunk 2]
```diff
@@ -1,17 +1,27 @@
 #!/usr/bin/env node
-const { Logger } = require('@hmcts/nodejs-logging')
+const appInsights = require('applicationinsights');
```
Comment (KiranVarma02) [inline src/main/server.ts]: `yarn add @types/applicationinsights` will add type defs and you can use import same like other imports.

File: src/main/server.ts [hunk 3]
```diff
@@ -1,17 +1,27 @@
...
+const config = require('@hmcts/properties-volume').addTo(require('config'));
+const { Logger } = require('@hmcts/nodejs-logging');
```
Comment (KiranVarma02) [inline src/main/server.ts]: you can do same if they have type defs if not this is fine.

**Review by KiranVarma02 — DISMISSED** [review]

File: src/main/modules/appinisght/index.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  if (config.has('applicationInsights.instrumentationKey')) {
+    appInsights.setup(config.get('applicationInsights.instrumentationKey'))
+      .setAutoCollectConsole(true, true)
+      .setDistributedTracingMode(appInsights.DistributedTracingModes.AI_AND_W3C)
+      .setSendLiveMetrics(true)
+      .start();
+  }
+};
+
+export {setup};
```
Comment (KiranVarma02) [inline src/main/modules/appinisght/index.ts]:
```suggestion
what do you think about same thing like this - 

export function setup (): void {
  if (config.has('applicationInsights.instrumentationKey')) {
    appInsights.setup(config.get('applicationInsights.instrumentationKey'))
      .setAutoCollectConsole(true, true)
      .setDistributedTracingMode(appInsights.DistributedTracingModes.AI_AND_W3C)
      .setSendLiveMetrics(true)
      .start();
  }
}

```

File: src/main/modules/secrets/index.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+const setup = (): void => {
+  if (config.has('secrets.adoption')) {
+    setSecret('secrets.adoption.AppInsightsInstrumentationKey', 'applicationInsights.instrumentationKey');
+  }
+};
+
+export {
+  setup,
+};
```
Comment (KiranVarma02) [inline src/main/modules/secrets/index.ts]:
```suggestion
function setSecret (secretPath: string, configPath: string): void {
  if (has(config, secretPath)) {
    set(config, configPath, get(config, secretPath));
  }
}

export function setup(): void {
  if (config.has('secrets.adoption')) {
    setSecret('secrets.adoption.AppInsightsInstrumentationKey', 'applicationInsights.instrumentationKey');
  }
}
```

Comment (tomaszpowroznik) [reply src/main/modules/secrets/index.ts]: done

File: src/main/modules/appinisght/index.ts (hunk 4 above)
Comment (tomaszpowroznik) [reply src/main/modules/appinisght/index.ts]: done

File: src/main/server.ts (hunk 1 above)
Comment (tomaszpowroznik) [reply src/main/server.ts]: done

File: src/main/server.ts (hunk 2 above)
Comment (tomaszpowroznik) [reply src/main/server.ts]: done

File: src/main/server.ts (hunk 3 above)
Comment (tomaszpowroznik) [reply src/main/server.ts]: done

File: src/main/app.ts [hunk 6]
```diff
@@ -5,14 +5,20 @@ import cookieParser from 'cookie-parser';
...
+import * as appinsight from './modules/appinisght';
 
-const config = require('config');
-const { Logger } = require('@hmcts/nodejs-logging');
+import config from 'config';
+
+secrets.setup(config);
+appinsight.setup(config);
+
+const {Logger} = require('@hmcts/nodejs-logging');
```
Comment (KiranVarma02) [inline src/main/app.ts:21]: i think this was from express-template and probably not part of your PR but while you are here, could you update this too please - `import { Logger } from '@hmcts/nodejs-logging';`

**Review by KiranVarma02 — APPROVED** [review]
LGTM ... added minor comment

**Pull request merged**